### ISSUE TITLE
Allow C/C++ libraries to have unit tests implemented in Swift

### ISSUE THREAD

**Issue by adammurdoch** [gradle/gradle-native#449]

**Comment by big-guy** [comment]
I think we should see how hard it would be to do https://github.com/gradle/gradle-native/issues/239 instead of the sneaky way we do things now for the XCTest and C++ unit test plugins, so we don't have the language * test framework explosion.

**Comment by adammurdoch** [comment]
Absolutely. We'd want some test framework agnostic way to ask for the "testable" variant of a particular binary. This variant should also not have a `main` symbol so that the consuming test suite doesn't need to know or care about this.
